nav2_waypoint_follower/src/waypoint_follower.cpp

```diff
@@ -305,13 +308,17 @@ void WaypointFollower::followWaypointsHandler(
       missedWaypoint.index = goal_index;
       missedWaypoint.goal = poses[goal_index];
       missedWaypoint.error_code = current_goal_status_.error_code;
+      missedWaypoint.error_msg = current_goal_status_.error_msg;
       result->missed_waypoints.push_back(missedWaypoint);
 
       if (stop_on_failure_) {
-        RCLCPP_WARN(
-          get_logger(), "Failed to process waypoint %i in waypoint "
-          "list and stop on failure is enabled."
-          " Terminating action.", goal_index);
+        result->error_code =
+          nav2_msgs::action::FollowWaypoints::Result::STOP_ON_MISSED_WAYPOINT;
+        result->error_msg =
+          "Failed to process waypoint " + std::to_string(goal_index) +
+          " in waypoint list and stop on failure is enabled."
+          " Terminating action.";
+        RCLCPP_WARN(get_logger(), result->error_msg.c_str());
         action_server->terminate_current(result);
         current_goal_status_.error_code = 0;
```

Comment:
Anywhere that ` current_goal_status_.error_code = 0;` --> ` current_goal_status_.error_msg = "";`. There are quite a few instances where its not properly reset